c_glib/test/test-record-batch.rb

```diff
@@ -67,19 +68,35 @@ def setup
                                              columns)
     end
 
-    def test_equal
-      fields = [
-        Arrow::Field.new("visible", Arrow::BooleanDataType.new),
-        Arrow::Field.new("valid", Arrow::BooleanDataType.new),
-      ]
-      schema = Arrow::Schema.new(fields)
-      columns = [
-        build_boolean_array([true, false, true, false, true]),
-        build_boolean_array([false, true, false, true, false]),
-      ]
-      other_record_batch = Arrow::RecordBatch.new(schema, 5, columns)
-      assert_equal(@record_batch,
-                   other_record_batch)
+    sub_test_case("#equal") do
+      def setup
+        require_gi_bindings(3, 4, 2)
+
+        @fields = [
+          Arrow::Field.new("visible", Arrow::BooleanDataType.new),
+          Arrow::Field.new("valid", Arrow::BooleanDataType.new),
+        ]
+        @schema = Arrow::Schema.new(fields)
+        @columns = [
+          build_boolean_array([true, false, true, false, true]),
+          build_boolean_array([false, true, false, true, false]),
+        ]
+        @record_batch = Arrow::RecordBatch.new(@schema, 5, @columns)
+      end
+
+      def test_equal
+        other_record_batch = Arrow::RecordBatch.new(@schema, 5, @columns)
+        assert_equal(@record_batch, other_record_batch)
+      end
+
+      def test_equal_metadata
+        schema_with_meta = @schema.with_metadata("key" => "value")
+        other_record_batch = Arrow::RecordBatch.new(schema_with_meta, 5, @columns)
+
+        assert @record_batch.equal_metadata(other_record_batch, false)
```

Comment:
```suggestion
        assert do
          @record_batch.equal_metadata(other_record_batch, false)
        end
```

This style shows more useful message on error.